application/client.plugins/projects/serial/src/lib/views/dialog/components.ts

```diff
@@ -52,21 +55,46 @@ export class SidebarVerticalPortDialogComponent implements OnInit, OnDestroy {
             if (typeof message !== 'object' && message === null) {
                 return;
             }
+            if (message) {
+                this._onSpyState(message);
+            }
             this._forceUpdate();
         });
-        this._ng_spyState = this._getSpyState();
         this._ng_ports = this._requestPortList();
         this._ng_ports.forEach(port => {
-            if (this._ng_spyState[port.path] === undefined ) {
+            if (this._ng_spyState[port.path] === undefined) {
                 this._ng_spyState[port.path] = 0;
             }
         });
+        this._interval = setInterval(() => {
+            this._subjects.event.next(true);
+        }, this._timeout);
     }
 
     ngOnDestroy() {
+        if (this._interval) {
+            clearTimeout(this._interval);
+        }
+        Object.keys(this._subscriptions).forEach((key: string) => {
+            this._subscriptions[key].unsubscribe();
+        });
         this._destroyed = true;
     }
 
+    public observe(): { event: Observable<any> } {
+        return { event: this._subjects.event.asObservable() };
+    }
```

Comment:
could you please name somehow events (onData, onPortData, onCloseDialog and so on). It allows better understand a code